### PR TITLE
MEN-6282: Install `mender-flash`, `mender-auth` and `mender-update` packages

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: scripts/test/run-tests.sh [hunk 1]
```diff
@@ -38,9 +38,9 @@ BBB_DEBIAN_EMMC_IMAGE_URL="https://rcn-ee.com/rootfs/bb.org/testing/2022-03-02/b
 ## Auto-update
 RASPBIAN_IMAGE_URL="https://downloads.raspberrypi.org/raspios_lite_armhf/images/raspios_lite_armhf-2022-09-26/2022-09-22-raspios-bullseye-armhf-lite.img.xz"
 
-UBUNTU_IMAGE_URL="https://downloads.mender.io/mender-convert/images/Ubuntu-Jammy-x86-64.img.gz"
+UBUNTU_IMAGE_URL="https://downloads.mender.io/mender-convert/images/wip-MEN-6282-Ubuntu-Jammy-x86-64.img.gz"
 
-DEBIAN_IMAGE_URL="https://downloads.mender.io/mender-convert/images/Debian-11-x86-64.img.gz"
+DEBIAN_IMAGE_URL="https://downloads.mender.io/mender-convert/images/wip-MEN-6282-Debian-11-x86-64.img.gz"
```
Comment (lluiscampos) [inline scripts/test/run-tests.sh:43]: Note to self. After review,
* [ ] Rename the images in S3 
* [ ] Drop this commit

**Comment by lluiscampos** [comment]
It depends on:
* https://github.com/mendersoftware/mender-image-tests/pull/92
* https://github.com/mendersoftware/mender/pull/1548

**Comment by lluiscampos** [comment]
@kacf @oleorhagen This long running PR is finally ready for review! :popcorn:

File: mender-convert-modify [hunk 2]
```diff
@@ -101,16 +101,30 @@ sudo mount ${root_part} work/rootfs
 log_info "Installing Mender client and related files"
 
 if [ "${MENDER_CLIENT_INSTALL}" = "y" ]; then
-    log_info "Installing Mender client version ${MENDER_CLIENT_VERSION}"
-    deb_get_and_install_package mender-client "${MENDER_CLIENT_VERSION}"
+    case "${MENDER_CLIENT_VERSION}" in
+        1* | 2* | 3*)
```
Comment (kacf) [inline mender-convert-modify:105]: Pedantic: Should use `1.* | ...` to avoid matching `10.0.0`.

File: mender-convert-modify [hunk 3]
```diff
@@ -101,16 +101,30 @@ sudo mount ${root_part} work/rootfs
...
     testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${DEB_NAME} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
 fi
 
 if [ "${MENDER_ENABLE_SYSTEMD}" == "y" ]; then
-    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-client.service \
-        work/rootfs/etc/systemd/system/multi-user.target.wants/mender-client.service"
+    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-auth.service \
+        work/rootfs/etc/systemd/system/multi-user.target.wants/mender-auth.service"
+    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-update.service \
+        work/rootfs/etc/systemd/system/multi-user.target.wants/mender-update.service"
```
Comment (kacf) [inline mender-convert-modify:132]: This one is unconditional. Shouldn't it test for version?

File: mender-convert-modify [hunk 4]
```diff
@@ -113,11 +114,15 @@ if [ "${MENDER_CLIENT_INSTALL}" = "y" ]; then
...
+            mender_client_deb_name="${DEB_NAME}"
+
+            log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
+            deb_get_and_install_package mender-flash "${MENDER_FLASH_VERSION}"
             ;;
     esac
 
     # Save installed client version for tests in Yocto variable format
-    testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${DEB_NAME} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
+    testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${mender_client_deb_name} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
```
Comment (kacf) [inline mender-convert-modify:125]: In both cases you are just setting `mender_client_deb_name` to `DEB_NAME`. So why introduce it?

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -10,11 +10,11 @@ variables:
     description: |-
       Version of mender-artifact for the builder image - 'master' or a specific version
   MENDER_CLIENT_VERSION:
-    value: "latest"
+    value: "master"
```
Comment (kacf) [inline .gitlab-ci.yml:13]: Commit message, did you actually mean: "The new packages won't land in `stable` APT distributions **until** after the release"?

File: mender-convert-modify [hunk 6]
```diff
@@ -101,16 +101,30 @@ sudo mount ${root_part} work/rootfs
...
+    esac
 
     # Save installed client version for tests in Yocto variable format
     testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${DEB_NAME} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
 fi
 
 if [ "${MENDER_ENABLE_SYSTEMD}" == "y" ]; then
-    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-client.service \
-        work/rootfs/etc/systemd/system/multi-user.target.wants/mender-client.service"
+    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-auth.service \
```
Comment (oleorhagen) [inline mender-convert-modify:129]: Q: Does this not have to be versioned?

File: tests/mender-image-tests
Comment (kacf) [inline tests/mender-image-tests:1]: Can the two mender-image-tests commits be squashed?

File: mender-convert-modify [hunk 7]
```diff
@@ -113,11 +114,15 @@ if [ "${MENDER_CLIENT_INSTALL}" = "y" ]; then
             log_info "Installing Mender Auth and Mender Update version ${MENDER_CLIENT_VERSION}"
             deb_get_and_install_package mender-auth "${MENDER_CLIENT_VERSION}"
             deb_get_and_install_package mender-update "${MENDER_CLIENT_VERSION}"
+            mender_client_deb_name="${DEB_NAME}"
+
+            log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
```
Comment (oleorhagen) [inline mender-convert-modify:119]: Q: Only for the newest client, or?

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -5,11 +5,26 @@ variables:
 
   # These variables are present elsewhere in the repository too. Make sure to
   # search for and change them too.
-  MENDER_ARTIFACT_VERSION: master
-  MENDER_CLIENT_VERSION: latest
-  MENDER_FLASH_VERSION: latest
-  MENDER_ADDON_CONNECT_VERSION: latest
-  MENDER_ADDON_CONFIGURE_VERSION: latest
+  MENDER_ARTIFACT_VERSION:
```
Comment (oleorhagen) [inline .gitlab-ci.yml:8]: Nice!

File: scripts/test/generate-image.sh [hunk 9]
```diff
@@ -66,9 +66,17 @@ cleanup_losetup() {
 generate_debian() {
     local -r image="${DEBIAN_IMAGE_ID}-x86-64.img"
 
-    mkosi --root-size=2G --distribution=debian --release="$DEBIAN_CODENAME" --format=gpt_ext4 --bootable --checksum \
-          --password password --package=openssh-server,dhcpcd5 --package grub-efi-amd64-signed \
-          --package shim-signed --package lsb-release --output="$image" build
+    mkosi --distribution=debian --release="$DEBIAN_CODENAME" --output="$image" \
```
Comment (oleorhagen) [inline scripts/test/generate-image.sh:69]: Q: why not have them all on each their own line now?

**Review by kacf — COMMENTED** [review]
Nice work!

**Review by oleorhagen — COMMENTED** [review]
Left a few comments regarding versioning and installing. Feel like I've missed something

File: mender-convert-modify (hunk 7 above)
Comment (lluiscampos) [reply mender-convert-modify:119]: It is for the new client. All clients 1.x, 2.x, 3.x will go in the previous case `1* | 2* | 3*)`.

File: mender-convert-modify (hunk 4 above)
Comment (lluiscampos) [reply mender-convert-modify:125]: Not very intuitive, I know.

`DEB_NAME` is defined in `deb_get_and_install_package` as a way for the user too know the file name that was actually installed. When installing the new client, we do install later `mender-flash` and then `DEB_NAME` gets overridden.

If you want, I can install first `mender-flash` and then the client packages and hence leave here `DEB_NAME`. Although that approach will make it weaker for future changes.

File: mender-convert-modify (hunk 6 above)
Comment (lluiscampos) [reply mender-convert-modify:129]: Yes. Good catch.

File: mender-convert-modify (hunk 3 above)
Comment (lluiscampos) [reply mender-convert-modify:132]: Yes. Good catch.

File: scripts/test/generate-image.sh (hunk 9 above)
Comment (lluiscampos) [reply scripts/test/generate-image.sh:69]: Okey, I'll do.

File: tests/mender-image-tests
Comment (lluiscampos) [reply tests/mender-image-tests:1]: Sure

**Review by oleorhagen — APPROVED** [review]

File: mender-convert-modify (hunk 4 above)
Comment (kacf) [reply mender-convert-modify:125]: No, it's fine, it just looked like it was done for no reason, and hence I suspected it was a leftover.

**Review by kacf — APPROVED** [review]

**Pull request merged**